core/modules/ccontrol/MergingHandler.h

```diff
@@ -45,6 +46,39 @@ namespace lsst {
 namespace qserv {
 namespace ccontrol {
 
+/// A class to delay creating the buffer until it is requested
+/// by xrootd SSI. The size of the buffer that will be needed is
+/// set using setTargetSize(int sz), and the buffer of that size is
+/// created by calling resizeToTargetSize(). When a buffer is no longer
+/// needed, zero() should be called to free the memory.
+class MergeBuffer {
+public:
+    using bufType = std::vector<char>;
+    MergeBuffer() {
+        _id = "MBI=" + std::to_string(_sequence++);
+        zero();
+    }
+    virtual ~MergeBuffer();
+
+    std::shared_ptr<bufType> getBuffer();
+    size_t getSize();
+    size_t getTargetSize() { return _targetSize; }
+    void setTargetSize(int sz);
+    void resizeToTargetSize();
```

Comment:
I don't think this should be the public method because the only use of the method is just before calling *getBuffer()*. You may consider doing an implicit resize as needed at a time when the pointer is requested. Besides, combining these two operations will make the code safer.